### PR TITLE
Feat: Add Hugging Face as LLM provider

### PR DESCRIPTION
**Pull request by lmilan**
### Description

Add docs for AI Proxy and AI Proxy Advanced configuration with Hugging Face.

[DOCU-4166](https://konghq.atlassian.net/browse/DOCU-4166)

### Testing instructions

Preview links:
- https://deploy-preview-8182--kongdocs.netlify.app/hub/kong-inc/ai-proxy/unreleased/how-to/llm-provider-integration-guides/huggingface/
- https://deploy-preview-8182--kongdocs.netlify.app/hub/kong-inc/ai-proxy-advanced/unreleased/how-to/llm-provider-integration-guides/huggingface/

### Checklist 

- [x] Review label added
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

[DOCU-4166]: https://konghq.atlassian.net/browse/DOCU-4166?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/_hub/kong-inc/ai-proxy-advanced/how-to/llm-provider-integration-guides/_huggingface.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Create the route:
+
+```bash
+curl -X POST http://localhost:8001/services/ai-proxy-advanced/routes \
+  --data "name=huggingface-chat" \
+  --data "paths[]=~/huggingface-chat$"
+```
+
+Enable and configure the AI Proxy Advanced plugin for Anthropic, replacing the `<huggingface_token>` with your own access token and `<huggingface_model>` with the name of the model to use.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-proxy-advanced/how-to/llm-provider-integration-guides/_huggingface.md:29]:
```suggestion
Enable and configure the AI Proxy Advanced plugin for Hugging Face, replacing the `<huggingface_token>` with your own access token and `<huggingface_model>` with the name of the model to use.
```

File: app/_hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/_huggingface.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+{% include_cached /md/plugins-hub/ai-providers-prereqs.md snippet='service' provider='Hugging Face' %}
+* Hugging Face access token with permissions to make calls to the Inference API
+* [Text-generation model](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending) from Hugging Face
+
+## Provider configuration
+
+### Set up route and plugin
+
+{% navtabs %}
+{% navtab Kong Admin API %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/_huggingface.md:22]: Why do we have the proxy advanced guides in `plugin_example` format, but the AI Proxy ones are written manually? I guess that's all of them, isn't it. Note to self to convert these examples after 3.9.

File: app/_hub/kong-inc/ai-proxy-advanced/overview/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-proxy-advanced/overview/_index.md:1]: There are two more tables on this page (for both plugins) that we should also update: request and response formats, and upstream URL patterns: https://deploy-preview-8182--kongdocs.netlify.app/hub/kong-inc/ai-proxy/unreleased/#request-and-response-formats

**Review by lena-larionova — CHANGES_REQUESTED** [review]

File: app/_hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/_huggingface.md (hunk 2 above)
Comment (lmilan) [reply app/_hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/_huggingface.md:22]: I followed the same structure as for the other providers, but I asked myself the same question

**Review by lena-larionova — APPROVED** [review]
LGTM!

**Pull request merged**